packages/flutter/test/material/switch_test.dart

```diff
@@ -187,4 +188,37 @@ void main() {
         ..circle(color: Colors.red[500])
     );
   });
+
+  testWidgets('Slider adapts to CupertinoSwitch on iOS', (WidgetTester tester) async {
+    bool switchValue = true;
+
+    await tester.pumpWidget(
+      new Material(
+        child: new Directionality(
+          textDirection: TextDirection.ltr,
+          child: new Theme(
+            data: new ThemeData(
+              adaptiveWidgetTheme: AdaptiveWidgetThemeData.bundled,
+              platform: TargetPlatform.iOS,
+            ),
+            child: new Switch(
+              value: switchValue,
+              onChanged: (bool value) => switchValue = value,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(
+      tester.widget<CupertinoSwitch>(
+        find.byType(CupertinoSwitch),
+      ).value,
+      true,
+    );
+
+    await tester.tap(find.byType(Switch));
+
+    expect(switchValue, false);
```

Comment:
these tests should all be checking that on Android the iOS adaptation doesn't happen

Reply:
specifically, what i'm looking for here is a test that "Switch" doesn't always use "CupertinoSwitch" regardless of platform, and so on. So it's a test for Switch, not a test for PlatformBuilder -- the test is checking that we use PlatformBuilder at all, in a sense.

Reply:
sure, done here and for the other widgets